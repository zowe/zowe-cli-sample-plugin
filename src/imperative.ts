/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {IImperativeConfig} from "@brightside/imperative";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    overrides: {
        CredentialManager: "./overrides/CredentialManager.override"
    },
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI sample plug-in",
    pluginAliases: ["zcsp"],
    rootCommandDescription: "Welcome to the sample plug-in for Zowe CLI!\n\n The sample plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Sample Plug-in",
    name: "zowe-cli-sample"
};

export = config;
