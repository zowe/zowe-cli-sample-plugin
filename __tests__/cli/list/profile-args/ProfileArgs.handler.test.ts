/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { mockHandlerParameters } from "@zowe/cli-test-utils";
import { IHandlerParameters, ImperativeConfig } from "@zowe/imperative";
import { ProfileArgsDefinition } from "../../../../src/cli/list/profile-args/ProfileArgs.definition";
import * as ProfileArgsHandler from "../../../../src/cli/list/profile-args/ProfileArgs.handler";

const DEFAULT_PARAMETERS: IHandlerParameters = mockHandlerParameters({
    positionals: ["brightside-sample-plugin", "list", "profile-args"],
    definition: ProfileArgsDefinition
});

describe("profile-args Handler", () => {
    afterAll(() => {
        jest.resetAllMocks();
    });

    it("should list profile args for team config", async () => {
        Object.defineProperty(ImperativeConfig, "instance", {
            get: () => ({
                config: {
                    api: {
                        layers: {
                            get: jest.fn()
                        },
                        secure: {
                            secureFields: jest.fn(() => [])
                        }
                    },
                    exists: true,
                    properties: {
                        defaults: {
                            base: "fakeBase",
                            sample: "fakeSample"
                        }
                    }
                },
                envVariablePrefix: "TEST",
                loadedConfig: {
                    profiles: []
                }
            })
        });
        const profileArgs = {
            host: "fakeHost",
            port: 443,
            user: "fakeUser",
            password: "fakePass",
            tshirtSize: "S"
        };

        const handler = new ProfileArgsHandler.default();
        const params = Object.assign({}, ...[DEFAULT_PARAMETERS]);
        params.arguments = {...params.arguments, ...profileArgs};

        const expectedOutput = {
            arguments: {
                ...profileArgs,
                rejectUnauthorized: true
            },
            environment: {
                usingTeamConfig: true,
                sampleProfileName: "fakeSample",
                baseProfileName: "fakeBase"
            }
        };
        let actualOutput = null;
        params.response.data.setObj.mockImplementation((obj: any) => {
            actualOutput = obj;
        });

        await handler.process(params);
        expect(actualOutput).toMatchObject(expectedOutput);
    });
});
