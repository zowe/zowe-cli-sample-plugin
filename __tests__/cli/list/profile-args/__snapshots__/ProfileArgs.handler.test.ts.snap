// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile-args Handler should list profile args for team config 1`] = `
Object {
  "format": "object",
  "output": Object {
    "arguments": Object {
      "host": "fakeHost",
      "password": "fakePass",
      "port": 443,
      "rejectUnauthorized": true,
      "tshirtSize": "S",
      "user": "fakeUser",
    },
    "environment": Object {
      "baseProfileName": "fakeBase",
      "sampleProfileName": "fakeSample",
      "usingTeamConfig": true,
    },
  },
}
`;
